app/src/main/java/at/tugraz/ist/swe/MainActivity.java

```diff
@@ -1,111 +1,120 @@
+
+public class MainActivity extends AppCompatActivity {
+
+    public List<Integer> tools = new ArrayList<>();
+    public LinearLayout layout;
+    public ColorPicker foreground;
+    public FlyoutMenuView toolFlyoutMenu;
+    public DrawArea drawingArea;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        tools.add(R.drawable.ic_outline_color_lens_24px);
+        tools.add(R.drawable.ic_si_glyph_circle);
+
+        layout=findViewById(R.id.main_canvas_view);
+
+
+        setupToolbar();
+
+        foreground = new ColorPicker(this);
+        foreground.setOnColorAppliedListener(new ColorPicker.ColorAppliedListener() {
+            @Override
+            public void onColorApplied(int color) {
+                drawingArea.getPaintingTool().setColor(color);
+            }
+        });
+
+        drawingArea = new DrawArea(this);
+
+        layout.addView(drawingArea);
+    }
+
+    private void setupToolbar() {
+        toolFlyoutMenu = findViewById(R.id.toolFlyoutMenu);
+
+        List<FlyoutToolbar.MenuItemImage> menuItemsImages = new ArrayList<>();
+
+        for (int item : this.tools)
+        {
+            menuItemsImages.add(new FlyoutToolbar.MenuItemImage(item, item, this.getApplicationContext()));
+        }
+        DisplayMetrics display;
+        display = this.getApplicationContext().getResources().getDisplayMetrics();
+
+        int width = toolFlyoutMenu.getItemWidth();
+        int margin = toolFlyoutMenu.getItemMargin() * 2;
+
+        int colums;
+        for (colums = 5; colums >= 0; --colums)
+        {
+            if(width * colums + margin * colums < display.widthPixels)
+                break;
+        }
+
+        toolFlyoutMenu.setLayout(new FlyoutMenuView.GridLayout(colums, FlyoutMenuView.GridLayout.UNSPECIFIED));
+        toolFlyoutMenu.setAdapter(new FlyoutMenuView.ArrayAdapter<>(menuItemsImages));
+
+        final FlyoutToolbar.ButtonRenderer renderer = new FlyoutToolbar.ButtonRenderer(R.drawable.ic_outline_apps_24px, this.getApplicationContext());
+        toolFlyoutMenu.setButtonRenderer(renderer);
+
+        toolFlyoutMenu.setSelectionListener(new FlyoutMenuView.SelectionListener() {
+            @Override
+            public void onItemSelected(FlyoutMenuView flyoutMenuView, FlyoutMenuView.MenuItem item) {
+
+                FlyoutToolbar.MenuItemImage selected = (FlyoutToolbar.MenuItemImage) item;
+
+                for(int counter = 0; counter < tools.size(); counter++)
+                {
+                    if(tools.get(counter) == selected.getID()) {
+                        showTool(selected.getID());
+                        break;
+                    }
+                }
+            }
+
+            @Override
+            public void onDismissWithoutSelection(FlyoutMenuView flyoutMenuView) {
+            }
+        });
+    }
+
+    @Override
+    public void onDestroy(){
```

Comment:
Don't make dlg_color public and access it directly. Instead implement a dismiss function in the color picker that can be called from the main activity which return the current selected color from the text box.

Reply:
Done.